### ISSUE TITLE
Provide a feature matrix CodiMD vs HackMD

### ISSUE THREAD

**Issue by johanricher** [hackmdio/codimd#1145]
I really applaud your work on CodiMD, which is IMHO the absolute best free/ibre open source pad, but as a CodiMD user and potential customer of HackMD, I want to know the differences in features between the different products that you develop and sell.

For example, as for right now, it's not possible to know if the "Book Mode" feature is available in CodiMD. (spoiler: it's not, which is really a shame)

Please look into making a feature matrix, similar as what Gitlab does (see their [Pricing](https://about.gitlab.com/pricing/), [Gitlab.com](https://about.gitlab.com/pricing/gitlab-com/feature-comparison/) and [Gitlab self hosted](https://about.gitlab.com/pricing/self-managed/feature-comparison/) feature pages). 

Here a simple two-columns comparison between CodiMD and HackMD would to the job.

**Comment by SISheogorath** [comment]
It sounds like an idea, even when I don't think it'll work out in a long run, since CodiMD is only maintained and developed by the community while HackMD is these days an own product making huge progress.

Since we started in late 2017 we started to develop into different directions and while the HackMD team is still proud of our work and tries to help for example by financing things like Stickers, we don't share any new feature since then, which makes it at some point a disjoined list.

I think one could create such a list, but to be honest I lost track of features on our as well as on their side. But a general feature list for CodiMD could definitely be a first step.

@jackycute you may want to have a look at this one and provide your input from HackMD side.

**Comment by johanricher** [comment]
Hi @SISheogorath thanks for your insights. Are you saying that CodiMD and hackmd.io don't share a codebase anymore? That seems weird to me, and not a good sign for CodiMD's future in my opinion. I assumed that CodiMD was the community edition of HackMD, **both** financed by selling services around the common/core software.

**Comment by ccoenen** [comment]
Yes, they are completely seperate projects now. It's not a CE / EE model.

**Comment by johanricher** [comment]
> they are completely seperate projects now

... but sharing the same Github account, website and even identity (or lack thereof, in CodiMD's case). Hence my confusion. Maybe CodiMD should have its own identity, then?

**Comment by SISheogorath** [comment]
@johanricher We (on community side) wanted to avoid the problem OwnCloud had with CE vs. EE. And also HackMD was already working a lot on their own way when the discussion came up.

A little bit of history:

https://github.com/hackmdio/codimd/issues/531
https://github.com/hackmdio/codimd/issues/579
https://github.com/hackmdio/codimd/issues/720

TL;DR: Historically grown ^^

If you prefer to talk a bit more interactive talking, there is also a [Matrix Channel](https://matrix.to/#/!ybaSlrrMNLeuoQuUuR:matrix.org?via=matrix.org&via=matrix.allmende.io&via=disroot.org).

**Comment by johanricher** [comment]
Alright this is cleared now, thanks for the explanations. Please consider moving to another Github account and adding a disclaimer on the HackMD website to avoid future confusion.

**Comment by ApoRes1337** [comment]
We will see maybe the WebDav Support for CodiMD ?

**Comment by ccoenen** [comment]
@ApoRes1337 there's a ticket about it: #245, but to my knowledge nobody is actively working on it.

**Comment by itamargiv** [comment]
I'm not sure if this is the correct address for it, but I'm also interested in the book mode feature, I was actually sifting through the issues (and even read through the whole CE-EE-fork saga) but could not find any open issue about it. I think it is a feature that is perhaps worth investing time on. Can I open a feature request on it? Is there any procedure to follow?

**Comment by almereyda** [comment]
We know of at least one project, Growi, that integrates the real-time editing interface into a wiki writing environment https://github.com/hackmdio/codimd/issues/969#issuecomment-430417512, if that's close enough to *book mode* for now.

**Comment by reagle** [comment]
Even without a matrix, CodiMD could at least provide a list of its features.

**Comment by Fonata** [comment]
Why is there active development on both repositories?
- https://github.com/codimd/server
- https://github.com/hackmdio/codimd

They have the same license.

I highly recommend updating the [CONTRIBUTING.md](https://github.com/hackmdio/codimd/blob/master/CONTRIBUTING.md) to explain why it is worthwhile that the community spends time on both - or better yet: let everyone know that this repo here will be set to read-only on a predefined date, like https://github.com/ariya/phantomjs. We can move all PRs and issues over to https://github.com/codimd/server.

**Comment by jackycute** [comment]
https://github.com/hackmdio/codimd (here) will be maintained by HackMD team.

**Comment by Fonata** [comment]
@jackycute Thanks for your quick response!

Is it correct to assume that the main advantage for the HackMD team is that this repo here has more predictability?

@ccoenen @jackycute I guess that keeping the repos in sync would be quite a lot of work and doesn't seem realistic. Let's say someone "from the street" finds the software useful and lacks a certain feature. What would be your advice where to open the issue or PR? In my opinion, the answer should be in [CONTRIBUTING.md](https://github.com/hackmdio/codimd/blob/master/CONTRIBUTING.md).

**Comment by ccoenen** [comment]
That's a long story, the short version is, both versions will probably develop further and further apart. Right now, they are mostly compatible, but I do not believe this will remain so forever.

The most important bits around that fork are probably in issues #1170, #720, https://github.com/codimd/server/issues/10 and also #1219. I tried to write this up in a mostly neutral way in https://github.com/codimd/server/blob/master/docs/history.md

**Comment by lapineige** [comment]
Hello,
Is there any update on that point ? I can't find any feature comparison.

**Comment by jackycute** [comment]
> Hello, Is there any update on that point ? I can't find any feature comparison.

We do have a draft for this, still WIP.
https://hackmd.io/@docs/hackmd-vs-codimd

**Comment by lapineige** [comment]
Great :)
Do you know if there is any translated version ? I can't read it 😅
